multithreading/task2/src/main/java/SumPrinter.java

```diff
@@ -0,0 +1,26 @@
+import java.util.List;
+
+public class SumPrinter implements Runnable {
+
+  private final List<Integer> nums;
+
+  public SumPrinter(List<Integer> nums) {
+    this.nums = nums;
+  }
+
+  @Override
+  public void run() {
+    while (true) {
+      synchronized (nums) {
```

Comment:
You need to add invocation of wait/notify methods everywhere where you use the nums list